### PR TITLE
feat: release cellxgene-schema to test pypi

### PR DESCRIPTION
**Pull request by nayib-jose-gloria**
- Set up to release new minor version to Test PyPI and PyPI when there are ontology/GENCODE reference file bumps in main
- For now, setting it to release as a release candidate for testing, and only to test PyPI. Will change once ready to bump in prod
- Added Test PyPI / PyPI API keys to Github secrets
- Test PyPI 3.1.0rc0 published release: https://test.pypi.org/project/cellxgene-schema/3.1.0rc0/
- Action triggering release above: https://github.com/chanzuckerberg/single-cell-curation/actions/runs/5125586931

### CODE REVIEW

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 3.0.2-rc.0
+current_version = 3.0.0-rc.1
```
Comment (Bento007) [inline .bumpversion.cfg:2]: Why did this reset back to 3.0.0?

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Build dist
+        run: >-
+          make pydist
+      - name: Publish distribution to Test PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository-url: https://test.pypi.org/legacy/
+          packages-dir: cellxgene_schema_cli/dist/
+# TODO: Uncomment once ready to release to prod PyPI
```
Comment (Bento007) [inline .github/workflows/publish-to-pypi.yml:40]: It feels odd to release to both pypi and testpypi at the same time. We shouldn't test that the RC works before release to pypi. We should install it using testpypi and run a few tests to confirm it works.

File: .bumpversion.cfg (hunk 1 above)
Comment (nayib-jose-gloria) [reply .bumpversion.cfg:2]: This was an artifact of testing, I'll fix

File: .github/workflows/publish-to-pypi.yml (hunk 2 above)
Comment (nayib-jose-gloria) [reply .github/workflows/publish-to-pypi.yml:40]: Unit tests already run against a local build from the same dist files we push. What other tests do you have in mind?

Comment (Bento007) [reply .github/workflows/publish-to-pypi.yml:42]: download from testpipy and make sure it runs correctly.

**Review by Bento007 — APPROVED** [review]

**Pull request merged**